### DISCUSSION TITLE
[A-L] (13 Febbraio 2024) Esercizio 5

### DISCUSSION THREAD

**Discussion by FeddyLix17** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#409] [Esami A-L]
La vostra automobile ha tre pulsanti:

- sul primo e sul secondo è scritto: "Questo pulsante non innesca l’autodistruzione”,

- mentre sul terzo è scritto: “Il primo pulsante innesca l’autodistruzione”.

---

1. Formalizzare le affermazioni scritte sui tre pulsanti in un adeguato linguaggio proposizionale. [1 punto]

2. Formalizzare l’informazione che uno e uno solo dei pulsanti innesca l’autodistruzione. [1 punto]

3. Formalizzare l’informazione che solo una delle scritte sui pulsanti è vera. [1 punto]

4. Usando le informazioni dei tre punti precedenti, si può dire quale pulsante innesca l’autodistruzione? Argomentare usando un metodo a piacere (tavole di verità, conseguenza logica, etc.) [2 punti]

> l'esercizio è uguale per entrambe le file

**Comment by DavideGalilei** [comment]
### 1. Tre pulsanti formalizzati

Definiamo il significato delle variabili:
- $\large{A}$: Il primo pulsante innesca l'autodistruzione (quando $\large{A = 1}$)
- $\large{B}$: Il secondo pulsante innesca l'autodistruzione (quando $\large{B = 1}$)
- $\large{C}$: Il terzo pulsante innesca l'autodistruzione (quando $\large{C = 1}$)

Il primo pulsante dice "Questo pulsante non innesca l’autodistruzione": $\large{\lnot{A}}$
Il secondo pulsante dice "Questo pulsante non innesca l’autodistruzione": $\large{\lnot{B}}$
Il terzo pulsante dice "Il primo pulsante innesca l’autodistruzione": $\large{A}$

<img src="https://github.com/user-attachments/assets/60da10ee-9b2a-4d77-a4e7-a030da63a7ec" width="300">

### 2. Uno e uno solo dei pulsanti innesca l’autodistruzione
Ci sono due approcci equivalenti per formalizzare questa affermazione:

#### Primo approccio
- Nessuna coppia di bottoni può essere vera allo stesso momento:
  $\large{\lnot(A \land B) \land \lnot(B \land C) \land \lnot(A \land C)}$
- Ciò però non basta: nel caso dove nessun bottone inneschi l'autodistruzione, questa formula non basta, quindi bisogna includere il caso dove almeno uno dei bottoni inneschi l'autodistruzione: $\large{(A \lor B \lor C)}$
- La congiunzione dei due: $\large{\lnot(A \land B) \land \lnot(B \land C) \land \lnot(A \land C) \land (A \lor B \lor C)}$

#### Secondo approccio
- Definire tre casi per partizioni:
  1. solo il Primo bottone innesca l'autodistruzione: $\large{(A \land \lnot{B} \land \lnot{C})}$
  2. solo il Secondo bottone innesca l'autodistruzione: $\large{(\lnot{A} \land B \land \lnot{C})}$
  3. solo il Terzo bottone innesca l'autodistruzione: $\large{(\lnot{A} \land \lnot{B} \land C)}$
- La loro congiunzione (con gli or): $\large{(A \land \lnot{B} \land \lnot{C}) \lor (\lnot{A} \land B \land \lnot{C}) \lor (\lnot{A} \land \lnot{B} \land C)}$

<img src="https://github.com/user-attachments/assets/e3f969c1-f074-4a33-85ad-59a636cba620" width="500">

### 3. Solo una delle affermazioni sui pulsanti è vera
Analogamente al punto 2, possiamo scrivere la seguente formula:
$\large{\lnot(\lnot{A} \land \lnot{B}) \land \lnot(\lnot{B} \land A) \land \lnot(\lnot{A} \land A) \land (\lnot{A} \lor \lnot{B} \lor A)} =$
$\large{\lnot(\lnot{A} \land \lnot{B}) \land \lnot(\lnot{B} \land A) \land \lnot(0) \land (1) =}$
$\large{\lnot(\lnot{A} \land \lnot{B}) \land \lnot(\lnot{B} \land A) \land 1 =}$
$\large{\lnot(\lnot{A} \land \lnot{B}) \land \lnot(\lnot{B} \land A) =}$
$\large{\lnot(\lnot{A} \land \lnot{B}) \land \lnot(\lnot{B} \land A) =}$
$\large{(A \lor B) \land (B \lor \lnot{A}) =}$ (De Morgan)
$\large{(A \land B) \lor (A \land \lnot{A}) \lor (B \land B) \lor (B \land \lnot{A}) =}$
$\large{(A \land B) \lor 0 \lor B \lor (B \land \lnot{A}) =}$
$\large{(A \land B) \lor B \lor (B \land \lnot{A}) =}$
$\large{(A \land B) \lor B =}$ (assorbimento)
$\large{B}$ (assorbimento)

<img src="https://github.com/user-attachments/assets/a8e925d7-9482-431f-92ac-72edfd1a3b7c" width="500">

### 4. Quale pulsante innesca l’autodistruzione?
Possiamo creare una tavola di verità imponendo i casi esplosivo/non esplosivo per ogni bottone, e controllare se riusciamo a capire quale bottone innesca l'autodistruzione.

| $\large{A}$ | $\large{B}$ | $\large{C}$ | $\large{\lnot{A}}$ | $\large{\lnot{B}}$ | $\large{A}$ | Solo un bottone esplosivo | Solo una affermazione vera |
|---|---|---|----|----|---|----------------------|------------------------|
| 0 | 0 | 0 | 1 | 1 | 0 | 0 | 0 |
| 0 | 0 | 1 | 1 | 1 | 0 | 1 | 0 |
| 0 | 1 | 0 | 1 | 0 | 0 | 1 | 1 |
| 0 | 1 | 1 | 1 | 0 | 0 | 0 | 1 |
| 1 | 0 | 0 | 0 | 1 | 1 | 1 | 0 |
| 1 | 0 | 1 | 0 | 1 | 1 | 0 | 0 |
| 1 | 1 | 0 | 0 | 0 | 1 | 0 | 1 |
| 1 | 1 | 1 | 0 | 0 | 1 | 0 | 1 |

<img src="https://github.com/user-attachments/assets/9423ae11-aa15-4b44-b420-9b7b171fd616" width="400">

Analizzando la tavola di verità:
1. Dobbiamo trovare le righe dove sia "Solo un bottone esplosivo" che "Solo una affermazione vera" sono $\large{1}$
2. C'è solo una riga che soddisfa entrambe le condizioni: $\large{A=0, B=1, C=0}$
3. Quindi il secondo pulsante ($\large{B}$) è quello che innesca l'autodistruzione

La risposta è: **Il secondo pulsante innesca l'autodistruzione**
